### DISCUSSION TITLE
🎇 Boxes Version 3.0: Looking for Testers

### DISCUSSION THREAD

**Discussion by tobias-kuendig** [OFFLINE-GmbH/oc-boxes-support#66] [Announcements]
We have been hard at work and have the next major version of Boxes ready for testing!

The new version includes:

* Revision system (Ref #2)
* References (Ref #62, @roojai)
* Optimized Multisite Support
* Dynamic Labels in Structure Section of the Boxes Editor
* Page Searchbox in the Page Section of the Boxes Editor
* Unpinned Sidebar in Boxes Editor (Ref #59, @sellscompany, @PubliAlex)

For the full changelog, see https://docs.boxes.offline.ch/getting-started/changelog.html

The new version is online at https://boxes.offline.ch/demo.html. 

If you have any feedback regarding the new features, feel free to comment here.

**Comment by eger-heed** [comment]
Hey, this is great work! Very happy with the speed and progress of this plugin!

But I'm having a couple of issues with revisions on version 3.1.3:

- Changes are not visible by users that are not logged in. Only an old version of the revised page is visible when logged out, but changes are visible when logged in. Turning off revisions with .env-variable resolves this. And yes, the page is moved from draft to published. I checked the DB and everything looked alright, so I thought maybe it was a cache issue, but since turning off revisions made the issue go away I'm not sure.
- I had some issues where exported pages were imported with elements in reverse order, not sure what happened there, but this happened across installations or two different sites.

**Reply by tobias-kuendig** [reply]
Thank you for your feedback!

> Changes are not visible by users that are not logged in.

Could you try again in v3.1.4? I have optimized the query that fetches the revision. If it still does not work, the `getCmsPageForUrl` method in Boxes' Controller would be the location to check for this issue. There the `$page` should be resolved to the published version for users without a session and to the current draft for users with a session. This distinction is done in the `scopeCurrent()` in the `HasRevisions` trait.

> I had some issues where exported pages were imported with elements in reverse order, not sure what happened there, but this happened across installations or two different sites.

The import/export feautre has become increasingly complex with the new revisions feature. I have added this problem to our backlog and will keep you posted as soon as a fix is available.
